internal/controllers/topology/cluster/desired_state.go

```diff
@@ -841,14 +841,6 @@ func computeMachineDeploymentVersion(s *scope.Scope, machineDeploymentTopology c
 		return currentVersion, nil
 	}
 
-	// At this point the control plane is stable (not scaling, not upgrading, not being upgraded).
```

Comment:
Should we really just drop this check? I would expect that we mark MDs as pending upgrade if we have >= maxMDUpgradeConcurrency rolling out

Reply:
If we already hit the concurrency limit then the MD will be marked as pending in l.779. In this case  `AllowUpgrade` will return false because we hit the concurrency limit. 

This check was initially only added as a safety but infact the case this check covers was already covered when we initially mark all current rolling machinedpeloyments in l.530. 

With concurrency support dropping this check becomes more important to have faster upgrades. Consider the case where we have 5 MDs (md-0 to md4) and the upgrade concurrency is 3. Initially md-0, md-1 and md-2 will pick up the upgrade. Now if md-0 finishes the upgrade md-3 will start upgrading. md-3 will start upgrading even while md-1 and md-2 have not finished the upgrade because it is within the concurrency limits.
If we keep this check, md-3 will only start upgrading after md-0, md-1 and md-2 all finish upgrading.

Reply:
I was not trying to suggest to keep the check as is but to change it so it checks for concurrency. (just to answer your last section)

But as you wrote, this is already done in l.779 so no need to do it again.

Thx for the explanation!